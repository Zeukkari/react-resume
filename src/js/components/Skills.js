import React from 'react'

import { Box, Heading, Text } from 'grommet'

import { ResumeSection, SectionTitle } from './common'

const Skills = ({ skills }) => {
  const renderItems = skills.map(item => {
    return (
      <Box key="item.name">
        <Heading level={3}>{item.name}</Heading>
        <Box direction="row">
          <Text>{item.level}</Text>
        </Box>
        <Text weight="bold" size="medium">
          Highlights
        </Text>
        <Box direction="row">
          {item.keywords.map(keyword => (
            <Text key={keyword}>{keyword}</Text>
          ))}
        </Box>
      </Box>
    )
  })

  return (
    <Box margin="none" pad="none" border="all">
      <ResumeSection>
        <SectionTitle>Skills</SectionTitle>
        <Box direction="column">{renderItems}</Box>
      </ResumeSection>
    </Box>
  )
}

Skills.defaultProps = {
  skills: [],
}

export { Skills }
