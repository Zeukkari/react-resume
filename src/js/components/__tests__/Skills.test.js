import React from 'react'
import renderer from 'react-test-renderer'

import {Skills} from '../Skills'

import defaultResume from './mockResume.json'

test('renders skills section', () => {
  const data = defaultResume.skills
  const component = renderer.create(<Skills skills={data} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
