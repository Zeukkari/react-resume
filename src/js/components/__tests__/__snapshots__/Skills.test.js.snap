// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders skills section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 eRCNqM"
>
  <div
    className="StyledBox-sc-13pk1d4-0 lhNMsi"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 jVpYfH"
      size="medium"
    >
      Skills
    </h1>
    <div
      className="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 IRSNj"
      >
        <h3
          className="StyledHeading-sc-1rdh4aw-0 gRvHvF"
        >
          Web Development
        </h3>
        <div
          className="StyledBox-sc-13pk1d4-0 ffeAql"
        >
          <span
            className="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Master
          </span>
        </div>
        <span
          className="StyledText-sc-1sadyjn-0 dndifk"
          size="medium"
        >
          Highlights
        </span>
        <div
          className="StyledBox-sc-13pk1d4-0 ffeAql"
        >
          <span
            className="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            HTML
          </span>
          <span
            className="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            CSS
          </span>
          <span
            className="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Javascript
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
